inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_opset1_to_legacy.cpp

```diff
@@ -50,22 +50,73 @@
 #include <ngraph/pass/constant_folding.hpp>
 #include <ngraph/pass/manager.hpp>
 
+#include <ngraph/pass/graph_rewrite.hpp>
+
 #include <memory>
 #include <vector>
 
 bool ngraph::pass::ConvertOpSet1ToLegacy::run_on_function(std::shared_ptr<ngraph::Function> f) {
-    ngraph::pass::Manager OpSet1ToLegacy;
+    ngraph::pass::Manager manager;
     std::vector<std::shared_ptr<ngraph::pass::PassBase> > transforms;
 
-#define NGRAPH_PASS(NAME, NAMESPACE) transforms.push_back(OpSet1ToLegacy.register_pass<NAMESPACE::NAME>());
-#include <transformations/convert_opset1_to_legacy/convert_opset1_to_legacy_tbl.hpp>
-#undef NGRAPH_PASS
+    manager.register_pass<ngraph::pass::ConstantFolding>();
+    manager.register_pass<ngraph::pass::ConvertReduceToPooling>();
+    manager.register_pass<ngraph::pass::ConvertMod>();
+    manager.register_pass<ngraph::pass::ConvertMinimum>();
+    manager.register_pass<ngraph::pass::ConvertSubtract>();
+    manager.register_pass<ngraph::pass::ConvertDivide>();
+    manager.register_pass<ngraph::pass::ConvertNegative>();
+    manager.register_pass<ngraph::pass::ConvertDepthToSpace>();
+    manager.register_pass<ngraph::pass::ConvertSpaceToDepth>();
+    manager.register_pass<ngraph::pass::ConvertConvolutions>();
```

Comment:
All these passes look capable to be run in a single pass through the function. You just didn't have time to reorganize or it is blocked by something?

Reply:
It requires deeper investigation because some of this transformations will be moved to CommonOptimizations, some of them have dependencies on execution order and some of them can be merged into single pass.